pygmt/base_plotting.py

```diff
@@ -1167,8 +1167,8 @@ def basemap(self, **kwargs):
 
         """
         kwargs = self._preprocess(**kwargs)
-        if not ("B" in kwargs or "L" in kwargs or "T" in kwargs):
-            raise GMTInvalidInput("At least one of B, L, or T must be specified.")
+        if not ("B" in kwargs or "L" in kwargs):
+            raise GMTInvalidInput("Frame or map_scale must be specified.")
```

Comment:
We actually have `Tm` and `Td`, but agree that the error message should be reworded.

https://github.com/GenericMappingTools/pygmt/blob/2128dd4133817a6c85a05b595907d3f086986d26/pygmt/base_plotting.py#L1120-L1121

```suggestion
        if not ("B" in kwargs or "L" in kwargs or "T" in kwargs):
            raise GMTInvalidInput("At least one of frame, map_scale, rose or compass must be specified.")
```

Not sure if we should have separate checks for  `"Td"` and `"Tm"` though, or if `"T"` is enough? Could you stress test this?

Reply:
@weiji14 I mentioned `Tm` and `Td` in #730 as what I assume  are the aliases being referred to with `T`. I'm not familiar with the `rose` and `compass` arguments and haven't managed to plot them on a map, but I have at least had them run without raising an error.

I passed:
`fig.basemap(region=[0, 10, 0, 10], frame="af", compass="x1/2/1")`

and 

`fig.basemap(region=[0, 10, 0, 10], frame="af", rose="x1/2/1")`

Both of these ran without an error (although I couldn't get the compass or rose to actually plot), but the `GMTInvalidInput` was raised when I deleted `frame`.

Reply:
See if this test case provides some inspiration (this was written quite a while ago before the compass/ross alias was done):

https://github.com/GenericMappingTools/pygmt/blob/e6d45ecef0a8b1646517342cb084e90c23107008/pygmt/tests/test_config.py#L61-L70

Basically something like `fig.basemap(compass="jBR+w5c+d-4.5") `

Reply:
@weiji14 I'll take a look at this tomorrow.

Also, is there any reason the tests use the GMT aliases instead of the PyGMT parameters when calling functions? I was to try and add tests for map_scale, compass, and rose arguments to make sure the error isn't raised when they're passed.

Reply:
Part of the reason is historical. We did a big purge of short aliases in #474 on the user-facing side, but haven't got around to fixing the backend tests. That said, there are some tests which do explicitly use short GMT aliases, but the one I linked above should not be one of them.

TLDR: Use long aliases unless there's a reason to use short ones.

Reply:
@weiji14 Thanks for the info!

Regarding stress testing GMT accepting Td/Tm values, is there any easy way to use the pygmt folder from my repo in the environment instead of v0.2.1 that I have in my conda environment, or is writing good tests the better solution here? I'm not opposed to writing tests,  but it's a lengthy process to have to wait for tests to run after pushing a commit.

Reply:
> is there any easy way to use the pygmt folder from my repo in the environment instead of v0.2.1 that I have in my conda environment

You can activate your conda environment, run `cond uninstall pygmt` to uninstall the v0.2.1 version, and then change into your pygmt folder, and reinstall pygmt using `pip install -e .`. The command installs pygmt to your environment in "editable/developing" mode. Now in the conda environment, you can use the pygmt master or any other branches.

> it's a lengthy process to have to wait for tests to run after pushing a commit.

You don't have to rely on the CI tests. Instead, you can run all tests locally by running `make test`, or run tests in a specific file by `pytest pygmt/tests/test_coast.py`.

Reply:
If you do `make install` (i.e. `pip install -e`), that will install `PyGMT` in 'editable' mode so any changes should work once you reload the `PyGMT` library or use `importlib.reload`. As for tests, you can run just one file at a time following https://github.com/GenericMappingTools/pygmt/blob/master/CONTRIBUTING.md#testing-your-code, @seisman just simplified things with #725:

```
pytest pygmt/tests/test_somefile.py
```

Reply:
Thanks for the help with this! It looks like compass and rose do not individually satisfy the requirement and the error is still raised. Working on updating my pull request.